### ISSUE TITLE
Feature Request: Attach apitrace to Already Running Processes

### ISSUE THREAD

**Issue by pinwhell** [apitrace/apitrace#897]
I propose a new feature for APITrace that would allow users to attach to already running processes without the need to restart them. Currently, APITrace primarily starts a trace from the launch of an application. This feature would enable attaching to running processes to facilitate debugging and analysis without disrupting the application's execution.

## Example pseudo usage:
```
apitrace trace --pid <process_id>
```

**Comment by jrfonseca** [comment]
I agree this would be nice in theory, but I'm afraid this is extremely hard (if not outright impossible) to achieve in practice.

To be precise, get a trace from the point apitrace is attached is doable, but that trace would be useless -- it would be incomplete and one would not be able to replay it.

**Comment by pinwhell** [comment]
By replay you mean, sending the exact sequence of calls to another computer, right when they happend? if so, then what are the exact complications from doing it from right inside the actual Injected/tracing process?

**Comment by jrfonseca** [comment]
> By replay you mean, sending the exact sequence of calls to another computer, right when they happend?

Yes.  That's how Apitrace is able to inspect graphics state and debug.

> if so, then what are the exact complications from doing it from right inside the actual Injected/tracing process?

That's possible, but it's not how Apitrace was designed.

Apitrace copes with the huge interface surface of APIs like OpenGL and Direct3D by  using automatic code generation, meaning a few folks (mostly me) need only to write a small amount of code to handle these APIs.

Start/stopping tracing mid execution or debugging live process would require an insane amount of hand-written code to understand/serialize the graphics state, and the army of developers required to write it does not exist.

There have been much such efforts in the past for OpenGL, some even forks of Apitrace, and most fail.   They quickly handle toy apps, but never get to the point to handle real life GL apps well.  See also https://github.com/apitrace/apitrace/issues/31

The only project I know that can do this for modern OpenGL / Direct3D is https://renderdoc.org/  I recommend I take a look.

To summarize, although this feature would be great in theory, it is unfeasible in practice, and therefore not a direction I'm willing to pursue or entertain.

**Comment by pinwhell** [comment]
Thank you for taking the time to clarify the intricacies of Apitrace and the challenges associated with implementing real-time tracing within the injected/tracing process. Your detailed explanation put exact enfasis on the complexities involved, and I appreciate your perspective on this matter, i fully understnad your position.
